language/move-lang/src/naming/translate.rs

```diff
@@ -616,10 +616,10 @@ fn base_type(context: &mut Context, ty: E::SingleType) -> N::BaseType {
                 NB::Apply(resource_opt, tn, base_types(context, tys))
             }
             Some(RT::BuiltinType) => {
+                let ty_args = base_types(context, tys);
                 let bn_ = N::BuiltinTypeName_::resolve(&n.value).unwrap();
-                let kind_ = bn_.kind();
-                let tn = sp(loc, NN::Builtin(sp(loc, bn_)));
-                NB::Apply(Some(sp(loc, kind_)), tn, base_types(context, tys))
+                // TODO: get rid of this return?
+                return NB::builtin(loc, bn_, ty_args);
```

Comment:
Remove the return by adding a function `builtin_` in BaseType that doesn't have the loc